mycroft/stt/deepspeech_stt.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import absolute_import, division
+
+from mycroft.stt import STT
+from mycroft.util.log import LOG
+from timeit import default_timer as timer
+from os.path import exists, join, dirname
+from mycroft.util.download import download
+import tarfile
+from time import sleep
+
+
+class DeepSpeechSTT(STT):
+    URL = "https://github.com/mozilla/DeepSpeech/releases/download/v0.1.0/" \
+          "deepspeech-0.1.0-models.tar.gz"
+
+    def __init__(self):
+        super(DeepSpeechSTT, self).__init__()
+        try:
+            from deepspeech.model import Model
+            self.Model = Model
+        except ImportError:
+            LOG.error("could not import deep speech, \n run pip install "
+                      "deepspeech")
+            raise
+
+        self.BEAM_WIDTH = self.config.get("beam_width", 500)
+        self.LM_WEIGHT = self.config.get("lm_weight", 1.75)
+        self.WORD_COUNT_WEIGHT = self.config.get("word_count_weight", 1.00)
+        self.VALID_WORD_COUNT_WEIGHT = self.config.get(
+            "valid_word_count_weight", 1.00)
+
+        # These constants are tied to the shape of the graph used
+        # (changing them changes the geometry of the first layer), so make
+        # sure you use the same constants that were used during training
+        self.N_FEATURES = self.config.get("n_features", 26)
+        self.N_CONTEXT = self.config.get("n_context", 9)
+
+        self.model = self.config.get("model", join(dirname(__file__),
+                                                   "deepspeech",
+                                                   "output_graph.pb"))
+        self.lm = self.config.get("lm", join(dirname(__file__),
+                                             "deepspeech",
+                                             "lm.binary"))
+        self.alphabet = self.config.get("alphabet", join(dirname(__file__),
+                                                         "deepspeech",
+                                                         "alphabet.txt"))
+        self.trie = self.config.get("trie", join(dirname(__file__),
+                                                 "deepspeech",
+                                                 "trie"))
+        self.downloaded = False
+        self.dl = None
+        auto_dl = self.config.get("auto_download", False)
```

Comment:
What is the case that we don't auto-download? I'd say unit tests maybe, but there are none for this class. Since this has to be explicitly configured, can we just assume that if the model doesn't exist we should fetch it?